File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        int n = nums.size();
        sort(nums.begin(), nums.end());
        int cnt = 1;
        int maxcnt = 1;
        for (int i = 1; i < n; i++) {
            if (nums[i] - nums[i - 1] == 1) {
                cnt++;
                maxcnt = max(maxcnt, cnt);
            } else if (nums[i] != nums[i - 1]) {
                cnt = 1;
            }
        }
        return maxcnt;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> st = new HashSet<>();
        for (int num : nums) {
            st.add(num);
        }
        int max = 0;
        for (int num : st) {
            if (!st.contains(num + 1)) {
                int cnt = 1;
                int x = num - 1;
                while(st.contains(x--)) {
                    cnt++;
                }
                max = Math.max(max, cnt);
            }
        }
        return max;
    }
}